Submodule libmariadb 467a193b399..fd23d6bf192:
diff --git a/libmariadb/cmake/FindGSSAPI.cmake b/libmariadb/cmake/FindGSSAPI.cmake
index 4862ec5..9ebda68 100644
--- a/libmariadb/cmake/FindGSSAPI.cmake
+++ b/libmariadb/cmake/FindGSSAPI.cmake
@@ -46,6 +46,29 @@ if(GSSAPI_LIBS AND GSSAPI_FLAVOR)
 
 else(GSSAPI_LIBS AND GSSAPI_FLAVOR)
 
+  find_package(PkgConfig)
+  IF(PKG_CONFIG_FOUND)
+    pkg_search_module(GSSAPI mit-krb5-gssapi)
+    IF(GSSAPI_FOUND)
+      set(GSSAPI_FLAVOR "MIT")
+    ELSE()
+      pkg_search_module(GSSAPI heimdal-gssapi)
+      IF(GSSAPI_FOUND)
+        set(GSSAPI_FLAVOR "HEIMDAL")
+      ENDIF()
+    ENDIF()
+    IF(GSSAPI_FOUND)
+      message(STATUS "Found GSSAPI: ${GSSAPI_LIBRARIES}")
+
+      set(GSSAPI_INCS ${GSSAPI_INCLUDE_DIRS} CACHE STRING "")
+      set(GSSAPI_LIBS ${GSSAPI_STATIC_LIBRARIES} CACHE STRING "")
+      set(GSSAPI_FLAVOR ${GSSAPI_FLAVOR} CACHE STRING "")
+
+      mark_as_advanced(GSSAPI_INCS GSSAPI_LIBS GSSAPI_FLAVOR)
+      RETURN()
+    ENDIF()
+  ENDIF(PKG_CONFIG_FOUND)
+
   find_program(KRB5_CONFIG NAMES krb5-config PATHS
      /opt/local/bin
      /usr/lib/mit/bin/
diff --git a/plugin/auth_gssapi/cmake/FindGSSAPI.cmake b/plugin/auth_gssapi/cmake/FindGSSAPI.cmake
index cc8e62d1ca7..ad4be435180 100644
--- a/plugin/auth_gssapi/cmake/FindGSSAPI.cmake
+++ b/plugin/auth_gssapi/cmake/FindGSSAPI.cmake
@@ -39,6 +39,29 @@ if(GSSAPI_LIBS AND GSSAPI_FLAVOR)
 
 else(GSSAPI_LIBS AND GSSAPI_FLAVOR)
 
+  find_package(PkgConfig)
+  IF(PKG_CONFIG_FOUND)
+    pkg_search_module(GSSAPI mit-krb5-gssapi)
+    IF(GSSAPI_FOUND)
+      set(GSSAPI_FLAVOR "MIT")
+    ELSE()
+      pkg_search_module(GSSAPI heimdal-gssapi)
+      IF(GSSAPI_FOUND)
+        set(GSSAPI_FLAVOR "HEIMDAL")
+      ENDIF()
+    ENDIF()
+    IF(GSSAPI_FOUND)
+      message(STATUS "Found GSSAPI: ${GSSAPI_STATIC_LIBRARIES}")
+
+      set(GSSAPI_INCS ${GSSAPI_INCLUDE_DIRS} CACHE STRING "" FORCE)
+      set(GSSAPI_LIBS ${GSSAPI_STATIC_LIBRARIES} CACHE STRING "" FORCE)
+      set(GSSAPI_FLAVOR ${GSSAPI_FLAVOR} CACHE STRING "" FORCE)
+
+      mark_as_advanced(GSSAPI_INCS GSSAPI_LIBS GSSAPI_FLAVOR)
+      RETURN()
+    ENDIF()
+  ENDIF(PKG_CONFIG_FOUND)
+
   find_program(KRB5_CONFIG NAMES krb5-config heimdal-krb5-config PATHS
      /opt/local/bin
      ONLY_CMAKE_FIND_ROOT_PATH               # this is required when cross compiling with cmake 2.6 and ignored with cmake 2.4, Alex
